### PR TITLE
Resource: Support policy set definition grouping

### PR DESCRIPTION
**Pull request by pilor**
Add support for policy set definition grouping in the "resource" command module.

Introduces a `--definition-groups` parameter to the `az policy set-definition create` and `az policy set-definition update` commands.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@Juliehzl  please help with the PR.

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,10 @@ Release History
 
 * Support Local context in acr task run
 
+**ARM**
+
+* Update azure-mgmt-resource package to use 6.0.0
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst]: Please also specify the api version you used for Resource Type `MGMT_RESOURCE_POLICY`.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 2]
```diff
@@ -1002,7 +1002,7 @@ def test_resource_policy_identity(self, resource_group, resource_group_location)
         self.cmd('policy assignment delete -n {pan} -g {rg}')
 
     @ResourceGroupPreparer(name_prefix='cli_test_policy_management_group')
-    @AllowLargeResponse()
+    @AllowLargeResponse(4096)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py]: Why do you want to specify object size for response here?

**Comment by Juliehzl** [comment]
LGTM. If CI pass, I will agree.

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 3]
```diff
@@ -156,6 +156,7 @@ def load_arguments(self, _):
         c.argument('description', help='Description of policy set definition.')
         c.argument('params', help='JSON formatted string or a path to a file or uri with parameter definitions.', type=file_type, completer=FilesCompleter())
         c.argument('definitions', help='JSON formatted string or a path to a file or uri containing definitions.', type=file_type, completer=FilesCompleter())
+        c.argument('definition_groups', min_api='2019-09-01', help='JSON formatted string or a path to a file or uri containing policy definition groups. Groups are used to organize policy definitions within a policy set.', type=file_type, completer=FilesCompleter())
```
[the remark this answers is not in the document]
Comment (pilor) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: They are complimentary but different.  --definition-groups defines groups that the indidividual definitions can be placed in to help categorize and provide richer metadata.  --definitions is where the actual list of policy definitions within the set resides and each definition can be placed in 0..n groups 

> "groupNames": [ "groupName1", "groupName2" ]

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 2 above)
Comment (pilor) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py]: When listing policies to ensure the modified policy is present it retrieves all built-in policies that Microsoft publishes.  This is a decent amount of content and was exceeding the default AllowLargeResponse size.  We are in the process of adding filters and better pagination to these APIs which should allow me to cut these response sizes down in the tests in a future change.

**Comment by pilor** [comment]
@Juliehzl CI checks have passed

**Pull request merged**